azurelinuxagent/ga/exthandlers.py

```diff
@@ -979,39 +978,42 @@ def report_ext_handlers_status(self, incarnation_changed=False):
                       is_success=False,
                       message=msg)
 
-    @staticmethod
-    def write_ext_handlers_status_to_info_file(vm_status):
-        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
-
-        agent_details = {
-            "agent_name": AGENT_NAME,
-            "current_version": str(CURRENT_VERSION),
-            "goal_state_version": str(GOAL_STATE_AGENT_VERSION),
-            "distro_details": "{0}:{1}".format(DISTRO_NAME, DISTRO_VERSION),
-            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
-            "python_version": "Python: {0}.{1}.{2}".format(PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO),
-            "crp_supported_features": [name for name, _ in get_agent_supported_features_list_for_crp().items()],
-            "extension_supported_features": [name for name, _ in get_agent_supported_features_list_for_extensions().items()]
-        }
-
-        # Convert VMStatus class to Dict.
-        data = get_properties(vm_status)
-
-        # The above class contains vmAgent.extensionHandlers
-        # (more info: azurelinuxagent.common.protocol.restapi.VMAgentStatus)
-        handler_statuses = data['vmAgent']['extensionHandlers']
-        for handler_status in handler_statuses:
-            try:
-                handler_status.pop('code', None)
-                handler_status.pop('message', None)
-                handler_status.pop('extension_status', None)
-            except KeyError:
-                pass
-
-        agent_details['extensions_status'] = handler_statuses
-        agent_details_json = json.dumps(agent_details)
-
-        fileutil.write_file(status_path, agent_details_json)
+    def write_ext_handlers_status_to_info_file(self, vm_status):
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE.format(self.protocol.get_incarnation()))
+        status_blob_data = self.protocol.get_status_blob_data()
+        data = dict()
+        if status_blob_data is not None:
+            data = json.loads(status_blob_data)
+
+        # Populating the fields that does not come from vm_status or status_blob_data
+        data["agentName"] = AGENT_NAME
```

Comment:
Any reason for changing the naming convention from underscore to Camel casing? 

Eg: `agent_name` to `agentName`

Reply:
The new logic uses the details directly from the status blob and poping out the fields that are not needed. The data in the blob is in Camel casing and for consistently reason I changed the fields that are manually populated in Camel casing as well.

Reply:
Cool thanks!